rust/lance-core/src/container/list.rs

```diff
@@ -0,0 +1,275 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use std::collections::LinkedList;
+
+use deepsize::DeepSizeOf;
+
+/// A linked list that grows exponentially. It is used to store a large number of
+/// elements in a memory-efficient way. The list grows by doubling the capacity of
+/// the last element when it's full, the capacity can be limited by the `limit`
+/// parameter. The default value is 0, which means no limit.
+#[derive(Debug, Clone, Default)]
+pub struct ExpLinkedList<T> {
+    inner: LinkedList<Vec<T>>,
+    len: usize,
+    // The maximum capacity of single node in the list.
+    // If the limit is 0, there is no limit.
+    // We use u16 to save memory because ExpLinkedList is used in
+    // such use cases.
```

Comment:
```suggestion
    // We use u16 to save memory because ExpLinkedList should not
    // be used if the limit is that large.
```
Not sure if I am interpreting this comment correctly but here is my attempt to clarify.